metadata-models/src/main/pegasus/com/linkedin/dataquality/DimensionScore.pdl

```diff
@@ -0,0 +1,52 @@
+namespace com.linkedin.dataquality
+
+/**
+ * Record to capture Dimension Score information
+ */
+record DimensionScore {
+  /**
+  * dimensionName attribute to capture predefined dimension names.
+  * Required field if this record is initialized
+  */
+  dimensionName: enum DimensionName {
+    ACCURACY
+    COMPLETENESS
+    CONSISTENCY
+    INTEGRITY
+    SEMANTIC_CORRECTNESS
+    UNIQUENESS
+    VALIDITY
+    TIMELINESS
+    RELIABILITY
+    DUPLICATION
+    PRECISION
+  }
```

Comment:
I am a bit worried of using a fixed enums list here for possible Dimensions. Similar approach has been used for the `OwnershipTypes` and it turned out to not be meeting requirements of everyone and did not allow for easy extension via [non-fork metadata model extension](https://datahubproject.io/docs/metadata-modeling/extending-the-metadata-model/#to-fork-or-not-to-fork). Since then the entire `OnwershipTypes` model has been changed to not use enums for types but rather handle dynamic list of OwnershipTypes (as entities on their own) and link to them using their urns from within the `Ownership` aspect.
 The overall commit introducing the dynamic `OwnershipTypes` feature can be found here I think:
 https://github.com/datahub-project/datahub/commit/ea92b86e6ab4cbb18742fb8db6bc11fae8970cdb
 Please note though that the change is so profound it's an ongoing effort and multiple changes have been done over time to bring the project closer to the usage of new approach while maintaining backward compatibility.
 Datahub populates the list of `OwnershipTypes` during the bootstrap with [this PR](https://github.com/datahub-project/datahub/pull/9308) introducing possibility to define user's own list of ownership types to bootstrap (we found this approach the best to define semi-static list fitting our custom purposes).

Reply:
Agree. We, for example, define multiple types of validity ( "is this zip code valid" and "is this value valid according to our business rules").

Reply:
Having comments for each symbol with at a minimum help us understand the unique value provided by each. I sense that there may be some conceptual overlap, so it will be good to be extremely clear in the comments what each will be used for.

Reply:
Hi all, taking a stab at providing some details on these dimensions with definitions

In 2013, DAMA United Kingdom produced a white paper proposing six core dimensions of data quality. These dimensions were widely recognized as the industry standard and include:
- **Accuracy:** The degree to which data correctly describes the ‘real world’ object or event being described.
- **Completeness**: The proportion of data stored against the potential for 100%. 
- **Consistency**: The absence of difference, when comparing two or more representations of a thing against a definition.
- **Uniqueness**: No entity instance (thing) will be recorded more than once based upon how that thing is identified.
- **Validity**: Data is valid if it conforms to the syntax (format, type, range) of its definition.
- **Timeliness**: The degree to which data represent reality from the required point in time.

Even today, if you search various research institutions and vendors, you will find these six dimensions are considered the de-facto industry standard. However, there is no universally established standard for measuring data quality dimensions, and some frameworks have expanded beyond these six. For example, [icedq.com](https://icedq.com/6-data-quality-dimensions#:~:text=The%20six%20data%20quality%20dimensions%20are%20Accuracy%2C%20Completeness%2C%20Consistency%2C,total%20of%2010%20DQ%20dimensions.) adds currency, conformity, integrity, and precision to create 10 dimensions. In 2020, the Data Management Association (DAMA) identified as many as 65 dimensions.

For a more detailed overview of the core six data quality dimensions, please refer to [THE SIX PRIMARY DIMENSIONS FOR DATA QUALITY ASSESSMENT](https://www.sbctc.edu/resources/documents/colleges-staff/commissions-councils/dgc/data-quality-deminsions.pdf).
 
This PR also includes five additional quality dimensions prioritized for our customer base. With the exception of Integrity most are placeholders based off some customers who have what I might considered very detailed level of data quality management. This noted, below are the definitions for each:
- **Integrity**: The accuracy and consistency of data over its entire lifecycle. It ensures that data is maintained in a manner that is complete, unaltered, and correct. Integrity encompasses various aspects such as referential integrity, entity integrity, and domain integrity, which collectively ensure that the data remains trustworthy and reliable. A common question, Is the data compliant to the referential integrity rules?
- **Semantic Correctness**: Semantic Correctness refers to the extent to which data accurately represents the meaning and context of the real-world entity or event it is intended to describe. It ensures that the data not only conforms to syntactical rules but also conveys the correct meaning and is interpretable in the intended context. **Example: Customer Data**:
    - *Correct*: A customer's address field contains "123 Main St, Springfield, IL 62701".
    - *Incorrect*: The address field contains "123 Main St, Springfield, IL 12345". While the syntax may be valid, the ZIP code does not align with the city and state, indicating a semantic error.
- **Reliability**: The degree to which data is consistently accurate, dependable, and available when needed for its intended use. Reliable data should produce the same results under consistent conditions and should be trustworthy for decision-making processes. The likelihood that data will be consistently accurate and available. **Example: Sales Data**:
  - *Reliable*: Daily sales reports consistently show accurate numbers that match the point-of-sale transactions without discrepancies.
  - *Unreliable*: Daily sales reports occasionally show significant discrepancies compared to point-of-sale transactions, causing confusion and mistrust in the data.
- **Duplication**: The presence of identical or nearly identical records within a dataset. Duplication can lead to inefficiencies, inaccuracies, and confusion, as multiple records may represent the same real-world entity. Ensuring that data is free from duplicates is crucial for maintaining its integrity and reliability.
- **Precision**: The level of detail and exactness in the data. It measures how finely data is expressed or stored and ensures that data values are as specific and detailed as necessary for the intended use. Precision is crucial for tasks that require exact measurements, calculations, or detailed descriptions.

Reply:
Removed the enum list and added support for dynamic DimensionNames